test/unit/util/mapbox.test.js

```diff
@@ -272,5 +278,313 @@ test("mapbox", (t) => {
+
+                //Post second event
+                now += (60 * 1000); // A bit later
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                t.equal(window.server.requests.length, 1);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.ok(reqBody.created, new Date(firstEvent).toISOString());
+
+                t.end();
+            });
+
+            t.test('does not POST appuserTurnstile event second time when clock goes backwards less than a day', (t) => {
+                let now = +Date.now();
+                const firstEvent = now;
+                t.stub(browser, 'now').callsFake(() => now);
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                //Post second event
+                now -= (60 * 1000); // A bit earlier
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                t.equal(window.server.requests.length, 1);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.ok(reqBody.created, new Date(firstEvent).toISOString());
+
+                t.end();
+            });
+
+            t.test('POSTs appuserTurnstile event when access token changes', (t) => {
+                config.ACCESS_TOKEN = 'pk.new.*';
+
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.equal(req.url, `${config.EVENTS_URL}?access_token=pk.new.*`);
+                t.equal(req.method, 'POST');
+                t.equal(reqBody.event, 'appUserTurnstile');
+                t.equal(reqBody.sdkVersion, version);
+                t.ok(reqBody.userId);
+
+                t.end();
+            });
+
+            t.test('POSTs appUserTurnstile event on next calendar day', (t) => {
+                let now = +Date.now();
+                t.stub(browser, 'now').callsFake(() => now);
+
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                now += ms25Hours; // Add a day
+                const tomorrow = now;
+                event.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.equal(req.url, `${config.EVENTS_URL}?access_token=key`);
+                t.equal(req.method, 'POST');
+                t.equal(reqBody.event, 'appUserTurnstile');
+                t.equal(reqBody.sdkVersion, version);
+                t.ok(reqBody.userId);
+                t.ok(reqBody.created, new Date(tomorrow).toISOString());
```

Comment:
This should be `t.equal`. (`t.ok` only checks that the first argument is truthy).  Also, should `reqBody` be `JSON.parse(req.requestBody)[1]` rather than `[0]`?

Reply:
A few other `t.ok(x, y)`s should be `t.equals` as well

Reply:
Good 👀 @anandthakker !

> should reqBody be JSON.parse(req.requestBody)[1] rather than [0]?

Actually `req` should be `window.server.requests[1]`. Found and fixed after the `t.equal` change.